### PR TITLE
feat: 🎸 Improved QueryObject to handle more fields

### PR DESCRIPTION
**Pull request by conglei**
🏆 Enhancements

The commit is to ensure the feature parity between frontend and backend `QueryOjbect`

### CODE REVIEW

File: packages/superset-ui-chart/src/types/Query.ts [hunk 1]
```diff
@@ -7,6 +8,18 @@ export interface QueryObject {
...
+  extras?: any;
+  timeseries_limit?: number;
+  timeseries_limit_metric?: Metric | null;
+  time_range?: string;
+  since?: string;
+  until?: string;
+  row_limit?: number;
+  order_desc?: boolean;
+  is_timeseries?: boolean;
+  prequeries?: any[];
```
Comment (williaster) [inline packages/superset-ui-chart/src/types/Query.ts]: can `prequeries`, `extras`, and `orderby` by something besides `any`?

File: packages/superset-ui-chart/src/query/buildQueryObject.ts [hunk 2]
```diff
@@ -12,8 +14,39 @@ function getGranularity(formData: ChartFormData): string {
...
 export default function buildQueryObject<T extends ChartFormData>(formData: T): QueryObject {
+  const extras = {
+    druid_time_origin: (formData as DruidFormData).druid_time_origin || '',
+    having: (formData as SqlaFormData).having || '',
+    having_druid: (formData as DruidFormData).having_druid || '',
+    time_grain_sqla: (formData as SqlaFormData).time_grain_sqla || '',
+    where: formData.where || '',
+  };
+
+  const orgGroupby = formData.groupby || [];
```
Comment (williaster) [inline packages/superset-ui-chart/src/query/buildQueryObject.ts]: I wasn't sure what `org` meant for these vars initially, could just extract like 

```javascript
const { columns = [], groupby = [] } = formData;

// OR
const groupbySet = new Set([ ...formData.columns, ...formData.groupby ]);
```

File: packages/superset-ui-chart/src/query/buildQueryObject.ts [hunk 3]
```diff
@@ -12,8 +14,39 @@ function getGranularity(formData: ChartFormData): string {
...
+  const orgColumns = formData.columns || [];
+  const groupbySet = new Set(orgGroupby.concat(orgColumns));
+  const limit = formData.limit ? Number(formData.limit) : 0;
```
Comment (williaster) [inline packages/superset-ui-chart/src/query/buildQueryObject.ts]: does `limit=0` have any weird query side effects?

Comment (conglei) [reply packages/superset-ui-chart/src/query/buildQueryObject.ts]: This is the original logic from `viz.py`:

`limit = int(form_data.get('limit') or 0)`

**Comment by conglei** [comment]
back too 100% !

**Comment by conglei** [comment]
should I merge this PR?

**Pull request merged**